reconcile/change_owners/change_owners.py

```diff
@@ -260,12 +264,18 @@ def is_change_admitted(
     # this is not admitted.
     # A change might be admitted if a user that has the restrictive change
     # type is an approver and adds an /good-to-test comment.
-    return all(
-        is_coverage_admitted(c, mr_author, good_to_test_approvers)
+
+    to_check = [
+        c
         for change in changes
         for dc in change.diff_coverage
         for c in dc.coverage
         if c.change_type_processor.restrictive
+    ]
+
+    return all(
+        is_coverage_admitted(to_check, mr_author, good_to_test_approvers)
+        for dc in to_check
```

Comment:
`dc` is not used in `is_coverage_admitted(to_check, mr_author, good_to_test_approvers)`.

It seems the new logic is the changes are admitted when all restrictive coverage origins are admitted, right?

Not sure what's `origin`, but this maybe clear, keep original `is_coverage_admitted` not changed.

```python
    restrictive_coverages = [
        c
        for change in changes
        for dc in change.diff_coverage
        for c in dc.coverage
        if c.change_type_processor.restrictive
    ]

    change_types_to_approve = {c.origin for c in restrictive_coverages}
    change_types_approved = {c.origin for c in restrictive_coverages if is_coverage_admitted(c, mr_author, good_to_test_approvers)}
    return change_types_to_approve == change_types_approved
```

May also need to update function doc.

Reply:
Great spot! Totally missed that... will update it